File: 53-MaximumSubarray/53-MaximumSubarray.py
# Last updated: 6/11/2026, 11:37:43 AM
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        currsum = 0
        i = 0
        maxsum = -10000
        while(i<len(nums)):
            currsum += nums[i]
            maxsum = max(maxsum,currsum)
            if(currsum<0):
                currsum = 0
            i = i+1
        return maxsum
    # def maxSubArray(self, nums: List[int]) -> int:
    #     res = nums[0]
    #     total = 0
    #     for n in nums:
    #         total += n
    #         res = max(res, total)
    #         if total < 0:
    #             total = 0
    #     return res

File: 53-MaximumSubarray/53-MaximumSubarray.cpp
// Last updated: 6/11/2026, 11:37:38 AM
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int curs = 0;
        int maxsum = nums[0];
        int n = nums.size();
        for(int i=0; i<n; i++){
            curs += nums[i];
            maxsum = max(maxsum, curs);
            if(curs<0){
                curs = 0;
            }
        }
        return maxsum;

    }
};